src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -1601,43 +1601,61 @@ def add_waf_managed_rule_set(cmd, client, resource_group_name, policy_name,
                                           rule_group_overrides=[rule_group_override] if rule_group_override is not None else [])  # pylint: disable=line-too-long
 
     for rule_set in waf_policy.managed_rules.managed_rule_sets:
-        if rule_set.rule_set_type == rule_set_type:
+        if rule_set.rule_set_type == rule_set_type and rule_set.rule_set_version == rule_set_version:
             for rule_override in rule_set.rule_group_overrides:
                 if rule_override.rule_group_name == rule_group_name:
+                    # Add one rule
                     rule_override.rules.extend(managed_rule_overrides)
                     break
             else:
+                # Add one rule group
                 if rule_group_override is not None:
                     rule_set.rule_group_overrides.append(rule_group_override)
             break
     else:
+        # Add new rule set
         waf_policy.managed_rules.managed_rule_sets.append(new_managed_rule_set)
 
     return client.create_or_update(resource_group_name, policy_name, waf_policy)
 
 
-def update_waf_managed_rule_set(cmd, instance, rule_set_type, rule_set_version, rule_group_name, rules=None):
+def update_waf_managed_rule_set(cmd, instance, rule_set_type, rule_set_version, rule_group_name=None, rules=None):
     """
     Update(Override) existing rule set of a WAF policy managed rules.
     """
-    ManagedRuleGroupOverride, ManagedRuleOverride = cmd.get_models('ManagedRuleGroupOverride', 'ManagedRuleOverride')
+    ManagedRuleSet, ManagedRuleGroupOverride, ManagedRuleOverride = \
+        cmd.get_models('ManagedRuleSet', 'ManagedRuleGroupOverride', 'ManagedRuleOverride')
 
     managed_rule_overrides = [ManagedRuleOverride(rule_id=r) for r in rules] if rules else None
 
     rule_group_override = ManagedRuleGroupOverride(rule_group_name=rule_group_name,
```

Comment:
If `rule_group_name` can be `None`, How to deal with this line?

Reply:
If `rule_group_name ` is None, `updated_rule_set` will be set to the target rule set in line if statement at line 1647 and for-loop get break at line 1648.

Then, the target rule set will be removed at line 1657 and a new MangedRuleSet will be appended to.

However, the `rule_group_overrides` property of new added ManagedRuleSet shoule contain an invalid ManagedRuleGroupOverride model if user don't provide --rules either.

If user prvode `--rules` without `--rule-group-name`, then the ManagedRuleGroupOverride model is invalid, server will reject this kind of request.

Reply:
Add validator for it